src/eiffel_graphql_api/graphql/db/database.py

```diff
@@ -30,20 +30,33 @@
 # pylint: disable=global-statement
 
 
+class ConfigurationError(Exception):
+    """Indicates a problem with a configuration setting."""
+
+
 def connect(mock):
     """Connect and wait for connection to MongoDB.
 
     :param mock: Whether the server should be mocked or not. Used for tests.
     :type mock: bool
     """
     global DATABASE, CLIENT
-    host = os.getenv("MONGODB_HOST", "localhost")
-    port = os.getenv("MONGODB_PORT", "27017")
-    username = os.getenv("MONGODB_USERNAME")
-    password = os.getenv("MONGODB_PASSWORD")
-    database_name = os.getenv("DATABASE_NAME", "this_is_not_correct")
-    replicaset = os.getenv("MONGODB_REPLICASET") or None
-    LOGGER.info("Connecting to %r:%r", host, port)
+    url = os.getenv("MONGODB_CONNSTRING")
+    if not url:
+        raise ConfigurationError(
+            "The required MONGODB_CONNSTRING environment variable was "
+            "unset or empty."
+        )
+    database_name = os.getenv("MONGODB_DATABASE")
+    if not database_name:
+        raise ConfigurationError(
+            "The required MONGODB_DATABASE environment variable was unset or empty."
+        )
+    safe_url = url
+    urlparts = urllib.parse.urlparse(url)
+    if urlparts.password:
+        safe_url = safe_url.replace(urlparts.password, "REDACTED")
```

Comment:
I'd prefer "standard" password masking characters such as asterisk ("*") or bullet ("●"), to avoid any "what password is actually used" confusion.

Reply:
Sure. Replaced `REDACTED` with `*****`.